### DISCUSSION TITLE
Moving from ASM6 to ca65 - unnamed labels

### DISCUSSION THREAD

**Discussion by MartsINY1** [cc65/cc65#2428] [Q&A]
In switching ASM6 to ca65, I wondered how I could reproduce something similar in a simple way (no need for labels name) - in ASM6, doing BEQ +++ will go to the next line where it has a +++, if doing BED +++++, it will go to the next +++++ - I mainly use a private memory to determine ASM phase of an object - with this value I jump to appropriate ASM phase:
```
    LDA PRV__METALL__ASM__PTRN,X
    BNE ++
    ;If Setting initial properties
    ;{
        ;whatever code
        -
        +
        RTS
    ;}
    ++
    CMP #$01
    BNE ++
    ;If opening
    ;{
        ;whatever code
        BEQ +
    ;}
    ++
    CMP #$01
    BNE ++
    ;If whatever
    ;{
        ;whatever code
        BEQ -
    ;}
    ++
    
    -
    +
    RTS
```
I would use the unnamed labels to easily branch to any RTS:
```
    LDA PRV__METALL__ASM__PTRN,X
    BNE ++
    ;If Setting initial properties
    ;{
        ;whatever code
:        RTS
    ;}
    ++
    CMP #$01
    BNE ++
    ;If opening
    ;{
        ;whatever code
        BEQ +
    ;}
    ++
    CMP #$01
    BNE ++
    ;If whatever
    ;{
        ;whatever code
        BEQ -
    ;}
    ++
    
:    RTS
```
But how to deal with the implementation I do with ++ without needing names for label? Since I would keep anonymous for RTS since that way I can ensure I always jump to a RTS doing so (just need to ensure it doesn't jump between banks). My goal is like with the ++ I did with ASM6, to avoid giving names to labels.

**Comment by colinleroy** [comment]
The the cc65 syntax is with ':' labels, and ':+' and ':-' targets :

```
...code...
bne :+                    <= will branch to anonymous label 1
...code...
:                              <= anonymous label 1
...code...
beq :-                     <= will branch to anonymous label 1
bne :+                    <= will branch to anonymous label 2
...code...
:                             <= anonymous label 2
...
```

You can also use ':++' or ':--' to skip one label, as in
```
... code...
bne :++                  <= will branch to anonymous label 2
... code ...
:                              <= anonymous label 1
...code...
bcc :+                     <= will branch to anonymous label 2
...code...
:                              <= anonymous label 2
...code...
```

See https://www.cc65.org/doc/ca65-6.html#ss6.2 for more information about the different types of labels

**Reply by MartsINY1** [reply]
Hi and thanks a lot for taking the time to answer.
However you explain how it works right? This I understand. With what you explain I do not see how to solve my initial question?

**Comment by colinleroy** [comment]
> Hi and thanks a lot for taking the time to answer. However you explain how it works right? This I understand. With what you explain I do not see how to solve my initial question?

Ah, I'm sorry, I think I don't understand your question then...

**Reply by MartsINY1** [reply]
Hi updated my question a little, if you have time, feel free to let me know if it's clearer so others don't get confused
